### PR TITLE
Source Stripe: expand tiers when syncing plans

### LINKED ISSUE #19905: Source Stripe: plan.tiers not available by default for Stripe API versions >2020-03-02
## Environment
- **Airbyte version**: 0.40.22
- **OS Version / Instance**: AWS EC2
- **Deployment**: docker-compose
- **Source Connector and version**: Stripe 0.1.40
- **Destination Connector and version**: Snowflake 0.4.40
- **Step where error happened**: Sync job

## Current Behavior
Previously, we were running against Stripe API version [2020-03-02](https://stripe.com/docs/upgrades#2020-03-02), but when updating to [2022-11-15](https://stripe.com/docs/upgrades#2022-11-15) the `tiers` property on plans isn't included in the response by default. This was changed in version [2020-08-27](https://stripe.com/docs/upgrades#2020-08-27). This means tiers will never be synced for plans when using the latest Stripe API version.

Either we'd have to go back to a previous Stripe API version, or use `expand[]=data.tiers` when requesting plans.

## Expected Behavior
Stripe plan tiers should sync.

## Steps to Reproduce
1. Set your Stripe account setting API version to anything >=2020-08-27.
2. Create a Stripe product/price using tiered pricing.
3. Sync plans using the Airbyte Stripe connector.
4. See tiers never sync.

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by visnup**
## What
Fixes #19905.

> Previously, we were running against Stripe API version [2020-03-02](https://stripe.com/docs/upgrades#2020-03-02), but when updating to [2022-11-15](https://stripe.com/docs/upgrades#2022-11-15) the tiers property on plans isn't included in the response by default. This was changed in version [2020-08-27](https://stripe.com/docs/upgrades#2020-08-27). This means tiers will never be synced for plans when using the latest Stripe API version.
> 
> Either we'd have to go back to a previous Stripe API version, or use expand[]=data.tiers when requesting plans.

## How
Includes the [tiers property](https://stripe.com/docs/api/prices/object#price_object-tiers) when syncing plans.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by visnup** [comment]
I'm a new contributor, so unsure what tests I can add and have left this so far as a draft PR.

I set this up to run on our Airbyte EC2 instance and it seems to work.

**Comment by marcosmarxm** [comment]
Sorry the delay @visnup I'm going to review your contribution!

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by visnup** [comment]
Ping on this PR. I think it's decently simple? Let me know if I can move it along.

**Comment by sashei** [comment]
Ping - would appreciate having this functionality as well. Cross-posting to Slack.

**Comment by marcosmarxm** [comment]
Sorry the long-long-long delay to get this merge @visnup and @sashei. 
@sh4sh can you take a look here?

**Comment by marcosmarxm** [comment]
@supertopher there was a change to run tests in community PRs? Seeing this error: https://github.com/airbytehq/airbyte/actions/runs/4113048675/jobs/7098711577

**Comment by sh4sh** [comment]
The [failing check](https://github.com/airbytehq/airbyte/actions/runs/4119260873/jobs/7112719921) is due to some changes in how acceptance tests are configured. We've started implementing high-strictness acceptance testing for GA connectors: https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference/#strictness-level

We also need an approving review from @airbytehq/gl-python

File: airbyte-integrations/connectors/source-stripe/source_stripe/streams.py
Comment (davydov-d) [inline airbyte-integrations/connectors/source-stripe/source_stripe/streams.py:421]: Could you please reflect this change in the stream schema as well? As far as I understand, it does not correspond to the docs. (at least 2 fields missing)

**Review by davydov-d — CHANGES_REQUESTED** [review]
Also, please bump the connector version in the Dockerfile and add a record to the changelog. Thanks!

**Review by bazarnov — APPROVED** [review]

**Review by davydov-d — APPROVED** [review]

**Comment by visnup** [comment]
Is there anything else I should do here?

**Comment by davydov-d** [comment]
@visnup the master branch has been updated, could you please merge it into your branch and resolve the conflicts?
afterwards, we'll fix the build and then we'll be able to proceed with your PR. Thanks!

**Comment by davydov-d** [comment]
hey @marcosmarxm can you please proceed with this PR? it needs the changelog to be updated. The build should be fixed this week I believe, then it can be published and merged

**Comment by sh4sh** [comment]
Everything is passing except changelog, @visnup could you add a new changelog entry for v3.1.0? I tried to add it but i don't have push access to your repo.

**Comment by sh4sh** [comment]
The connector is now published, but I needed to bump connector versions in https://github.com/airbytehq/airbyte/pull/23941. Once we get an approving review, it will be merged in that pr.

**Comment by marcosmarxm** [comment]
This has been published one month ago. My recommentation is to delete the tag in Dockerhub and republish the connector again.

**Pull request merged**
